src/main/java/tasks/Task.java

```diff
@@ -0,0 +1,97 @@
+package tasks;
+
+/**
+ * tasks.Task class to represent a task.
+ *
+ * @author: Chen Hsiao Ting
+ */
+
+public class Task {
+    private String description;
+    private boolean isDone;
+    private String type;
+
+    /**
+     * A constructor for the task.
+     *
+     * @param description user input task description.
+     * @param isDone status of the task.
+     * @param type type of task.
+     */
+    public Task(String description, Boolean isDone, String type) {
+        this.description = description;
+        this.isDone = isDone;
+        this.type = type;
+    }
+
+    /**
+     * Returns the status of the task.
+     *
+     * @return a status in string.
+     */
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+
+    /**
+     * Returns the status and description of the task.
+     *
+     * @return a string representation of the task.
+     */
+    public String getTask() {
+        return "[" + getStatusIcon() + "] " + description;
+    }
+
+    /**
+     * Sets the task to done by marking its status.
+     */
+    public void done() {
```

Comment:
May want to differentiate the name of this method and the one below it for clarity.